### ISSUE TITLE
Failing Travis CI build

### ISSUE THREAD

**Issue by agitter** [gitter-lab/ssps#24]
The latest Travis CI job failed.  Part of the error message is
```
rule run_mcmc:
    input: ../SSPS/ssps_wrapper.jl, example_timeseries.csv, example_prior.csv
    output: temp/3.json
    jobid: 6
    wildcards: chain=3
    resources: runtime=70, threads=1, mem_mb=2000
Invoking SSPS on input files:
	example_timeseries.csv
	example_prior.csv
Sampling for 60.0 seconds. (Or 100000 iterations.)
ERROR: LoadError: MethodError: no method matching ##lambda_vec_proposal#450(::Gen.GFProposeState, ::getfield(Main.SSPS, Symbol("##StaticIRTrace_vertex_lambda_dbn_model#445")), ::Int64, ::Float64)
Closest candidates are:
  ##lambda_vec_proposal#450(::Any, ::Any, ::Int64, !Matched::Int64) at /home/travis/build/gitter-lab/ssps/SSPS/src/dbn_proposals.jl:246
```

Comparing the [last successful job](https://travis-ci.com/github/gitter-lab/ssps/jobs/371385764) with the [failed job](https://travis-ci.com/github/gitter-lab/ssps/jobs/470620691) there are some differences in installed packages:
```
$ diff sorted_success.txt sorted_failure.txt
1,2c1,3
<  Installed ArgParse ──────────────────── v1.1.0
<  Installed BinDeps ───────────────────── v1.0.1
---
>  Installed ArgParse ──────────────────── v1.1.1
>  Installed Arpack ────────────────────── v0.3.2
>  Installed BinDeps ───────────────────── v1.0.2
5,6c6,7
<  Installed CSV ───────────────────────── v0.7.7
<  Installed CategoricalArrays ─────────── v0.8.1
---
>  Installed CSV ───────────────────────── v0.8.2
>  Installed CategoricalArrays ─────────── v0.8.3
9,11c10,12
<  Installed Compat ────────────────────── v2.2.0
<  Installed DataAPI ───────────────────── v1.3.0
<  Installed DataFrames ────────────────── v0.21.6
---
>  Installed Compat ────────────────────── v2.2.1
>  Installed DataAPI ───────────────────── v1.4.0
>  Installed DataFrames ────────────────── v0.21.8
14,18c15,19
<  Installed DiffResults ───────────────── v1.0.2
<  Installed DiffRules ─────────────────── v1.0.1
<  Installed Distributions ─────────────── v0.23.9
<  Installed FillArrays ────────────────── v0.9.2
<  Installed ForwardDiff ───────────────── v0.10.12
---
>  Installed DiffResults ───────────────── v1.0.3
>  Installed DiffRules ─────────────────── v1.0.2
>  Installed Distributions ─────────────── v0.22.6
>  Installed FillArrays ────────────────── v0.8.14
>  Installed ForwardDiff ───────────────── v0.10.14
22c23
<  Installed Gen ───────────────────────── v0.3.5
---
>  Installed Gen ───────────────────────── v0.4.1
26,27c27,28
<  Installed JSON ──────────────────────── v0.21.0
<  Installed LRUCache ──────────────────── v1.1.0
---
>  Installed JSON ──────────────────────── v0.21.1
>  Installed LRUCache ──────────────────── v1.2.0
29c30
<  Installed LibExpat ──────────────────── v0.5.0
---
>  Installed LibExpat ──────────────────── v0.5.0
31,35c32,36
<  Installed MacroTools ────────────────── v0.5.5
<  Installed Missings ──────────────────── v0.4.3
<  Installed NaNMath ───────────────────── v0.3.4
<  Installed OrderedCollections ────────── v1.3.0
<  Installed PDMats ────────────────────── v0.10.0
---
>  Installed MacroTools ────────────────── v0.5.6
>  Installed Missings ──────────────────── v0.4.4
>  Installed NaNMath ───────────────────── v0.3.5
>  Installed OrderedCollections ────────── v1.3.2
>  Installed PDMats ────────────────────── v0.9.12
37c38,39
<  Installed Parsers ───────────────────── v1.0.10
---
>  Installed Parameters ────────────────── v0.12.1
>  Installed Parsers ───────────────────── v1.0.15
39c41
<  Installed QuadGK ────────────────────── v2.4.0
---
>  Installed QuadGK ────────────────────── v2.4.1
41c43
<  Installed ReverseDiff ───────────────── v1.4.2
---
>  Installed ReverseDiff ───────────────── v1.5.0
43c45
<  Installed SentinelArrays ────────────── v1.2.10
---
>  Installed SentinelArrays ────────────── v1.2.16
46,48c48,51
<  Installed StaticArrays ──────────────── v0.12.4
<  Installed StatsBase ─────────────────── v0.33.0
<  Installed StatsFuns ─────────────────── v0.9.5
---
>  Installed StaticArrays ──────────────── v1.0.1
>  Installed StatsBase ─────────────────── v0.32.2
>  Installed StatsFuns ─────────────────── v0.9.6
>  Installed StructTypes ───────────────── v1.2.1
50c53
<  Installed Tables ────────────────────── v1.0.5
---
>  Installed Tables ────────────────────── v1.2.2
52a56
>  Installed UnPack ────────────────────── v1.0.2
```

**Comment by dpmerrell** [comment]
What I've tried: committing a `Manifest.toml` file containing older versions of some packages. 

Result: Travis-CI passes on the Julia 1.4 jobs and fails on the Julia 1.2 jobs.

The Julia 1.2 failures come from at least one package (`OpenSpecFun_jll`) being incompatible with Julia 1.2:
```
ERROR: LoadError: Unable to import OpenSpecFun_jll on Julia versions older than 1.3!
```

Proposed solution: Discard our Julia 1.2 tests. 

* It's difficult to manually debug dependencies in Manifest.toml; the file is machine-generated. 
* Our README instructions point specifically to Julia 1.4. We originally included Julia 1.2 in travis for reasons that no longer apply (it happened to be the stable version when I started implementing SSPS).
* I can't imagine a reason for a new user to prefer SSPS in Julia 1.2 over Julia 1.4. 

@agitter let me know if you have opinions about this.

**Comment by agitter** [comment]
Dropping Julia 1.2 support and tests makes sense to me.  The constraint comes from an external package, and there isn't much we can do about that.

If you think SSPS is compatible with Julia 1.3, we could modify the 1.2 tests to 1.3 tests.  I don't have a strong opinion about that versus only testing in 1.4.
